docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -0,0 +1,350 @@
+    ```bash
+      const char *pipeline = "videotestsrc num_buffers=2 ! videoconvert ! videoscale ! video/x-raw,format=RGBx,width=224,height=224 ! tensor_converter ! fakesink";
+      ml_pipeline_h handle;
+      int status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+    ```
+
+2. Start the pipeline and get state.
+    ```bash
+      /* a pipeline could be started when the state is paused */
+      status = ml_pipeline_start (handle);
+
+      status = ml_pipeline_get_state (handle, &state);
+    ```
+
+3. Stop the pipeline
+    ```bash
+      status = ml_pipeline_stop (handle);
+      g_usleep (50000); /* 50ms. Let a few frames flow. */
+
+      status = ml_pipeline_get_state (handle, &state);
+    ```
+
+4. Pipeline destroy
+    After using it, the constructed pipeline have to be destroyed.
+    ```bash
+      status = ml_pipeline_destroy (handle);
+    ```
+
+### Element API <a name="elementapi"></a>
+To run deep learning models with Machine Learning Inference C-API, manipulating the input/output data is required. Furthermore, the constructed pipeline should be controllable.
+
+Below element APIs make it available.
+
+* #### Source <a name="sourceelement"></a>
+
+  In case of setting the input tensor data, the configuration of data source element is required.
+  ```bash
+    gchar * pipeline = g_strdup ("appsrc name=srcx ! other/tensor,dimension=(string)4:1:1:1,type=(string)uint8,framerate=(fraction)0/1 ! tensor_sink");
+  ```
+  Through `ml_pipeline_src_get_handle ()`, `appsrc` element with the name `srcx` become controllable.
+
+  ```bash
+    ml_pipeline_h handle;
+    ml_pipeline_src_h srchandle;
+
+    status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+    status = ml_pipeline_start (handle);
+    status = ml_pipeline_src_get_handle (handle, "srcx", &srchandle);
+  ```
+
+  With `srchandle`, we can check the information of input tensors.
+
+  ```bash
+    ml_tensors_info_h info;
+
+    status = ml_pipeline_src_get_tensors_info (srchandle, &info);
+  ```
+
+  According to the `info`, the input tensor data can be filled.
+  ```bash
+    ml_tensors_data_h data;
+    
+    status = ml_tensors_data_create (info, &data);
+
+    for (i = 0; i < 10; i++) {
+      uintarray1[i] = (uint8_t *) g_malloc (4);
+      uintarray1[i][0] = i + 4;
+      uintarray1[i][1] = i + 1;
+      uintarray1[i][2] = i + 3;
+      uintarray1[i][3] = i + 2;
+    }
+
+    status = ml_tensors_data_set_tensor_data (data, 0, uintarray1[0], 4);
+
+    /* setiing the policy of raw data pointer */
+    status = ml_pipeline_src_input_data (srchandle, data, ML_PIPELINE_BUF_POLICY_DO_NOT_FREE);
+  ```
+
+  After configuring it, release the handle is required.
+
+  ```bash
+    status = ml_pipeline_src_release_handle (srchandle);
+  ```
+
+* #### Sink <a name="sinkelement"></a>
+  In case of getting the output tensor raw data, the configuration of data sink element is required.
```

Comment:
Please rephrase to:
The configuration of the data sink element is required to get the output tensor raw data: